iox-#743 Validation of icedelivery example (#1210)
eclipse-iceoryx/iceoryx: iceoryx_examples/icedelivery/README.md

## Introduction

This example showcases a data transmission setup with zero-copy inter-process communication (IPC) on iceoryx.

> It provides publisher and subscriber applications. They come in two C++ API flavors (untyped and typed).

Reviewer: Technically speaking both versions are correct :smile:

Author: Correct, but I like consistency ;) (see line 14)

## Code walkthrough

This example makes use of two kinds of API flavors. With the untyped API, you have the most flexibility. It enables you

> to put APIs on a higher level with a different look and feel on top of iceoryx, e.g. the ara::com API of AUTOSAR Adaptive or

Reviewer 2: ```suggestion
to define higher level APIs with a different look and feel on top of iceoryx, e.g. the ara::com API of AUTOSAR Adaptive or
```

data->z = ct;
```

> And finally, the value is made available to other subscribers with

Reviewer 2: ```suggestion
Finally the value is made available to any subscriber with
```

<!--[geoffrey][iceoryx_examples/icedelivery/iox_publisher_untyped.cpp][publish]-->
```cpp
publisher.publish(userPayload);
```

> The incrementation and sending of the data is done in a loop every second until the user presses `Ctrl-C`. It is

Reviewer: [Same here](https://github.com/eclipse-iceoryx/iceoryx/pull/1210/files#r818117728)

> The incrementation and sending of the data is done in a loop every second until the user presses `Ctrl-C`. It is

Reviewer 2: ```suggestion
Incrementing the counter and sending of the data happens in a loop every second until the user presses `Ctrl-C`. It is
```

}
```

> The program execution is stopped when the user presses `Ctrl-C`.

Reviewer 3: We have to solve the issue if we are writing markdown documentation or mkdocs documentation for the website.

In my opinion this is github markdown documentation and therefore we should not enable those features.

But if we write mkdocs documentation for the website this is a nice feature, but then we should NOT directly link or provide the documentation via github and only direct to the website.

In the end the starting point of most developers will be github so I would see this is prio.

> The program execution is stopped when the user presses `Ctrl-C`.

Reviewer: I understand what you mean. However, for `v1.0` we decided to include the `.md` of all examples to avoid duplication work. Those `.md` files can be read via GitHub web GUI and as well on [iceoryx.io](https://iceoryx.io). We basically killed two birds with one stone.

I think most developers use iceoryx.io](https://iceoryx.io) to access the docs instead of viewing them via GitHub. I can dig out the click rates for both [iceoryx.io](https://iceoryx.io) and the access via GitHub if you're interested.

I would appreciate if we use the key feature, but it's not a blocker IMHO.

#### #4 Publish the result of a computation

Usage #4 can be useful if you have a callable, e.g. a function or
[functor](https://en.wikipedia.org/wiki/Function_object#In_C_and_C++) should be always called. The callable needs

> to have the signature `void(SampleType*, ...)`.  What then happens, is the following: The publisher loans a sample from

Reviewer 2: ```suggestion
to have the signature `void(SampleType*, ...)`.  The semantics are as follows: The publisher loans a sample from
```
